backend-shared/exportfuncs.cpp

```diff
@@ -35,23 +35,33 @@ bool ExportCallback::canceled() const
 
 #include "desktop-widgets/mainwindow.h" // Currently needed for profile printing. TODO: remove.
 
-static void exportProfile(const struct dive *dive, const QString filename)
+static void exportProfile(ProfileWidget2 *profile, const struct dive *dive, const QString &filename)
 {
-	ProfileWidget2 *profile = MainWindow::instance()->graphics;
-	profile->setToolTipVisibile(false);
-	profile->setPrintMode(true);
-	double scale = profile->getFontPrintScale();
-	profile->setFontPrintScale(4 * scale);
 	profile->plotDive(dive, 0, false, true);
 	QImage image = QImage(profile->size() * 4, QImage::Format_RGB32);
 	QPainter paint;
 	paint.begin(&image);
 	profile->render(&paint);
 	image.save(filename);
+}
+
+static ProfileWidget2 *getPrintProfile()
+{
+	ProfileWidget2 *profile = MainWindow::instance()->graphics;
```

Comment:
that's still just the one in the `MainWindow`. Here's a crazy thing. If you execute the main event loop during printing (to make sure that the progress bar actually moves forward), you can make see all the profiles rendered into the visible profile window as you print. I had assumed that you would also take a 'private' profile widget here...

Maybe I'm confused?